### PR TITLE
Flink: Migrate subclasses of FlinkCatalogTestBase to JUnit5

### PR DESCRIPTION
**Pull request by vinitpatni**
- Added junit 5 conversion and AssertJ style for TestFlinkCatalogTable and TestFlinkMetadataTable

### CODE REVIEW

File: flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/TestFlinkCatalogTable.java [hunk 1]
```diff
@@ -94,84 +89,77 @@ public void testGetTable() {
...
+        .isEqualTo(Sets.newHashSet(table.schema().findField("key").fieldId()));
     CatalogTable catalogTable = catalogTable("tl");
     Optional<UniqueConstraint> uniqueConstraintOptional = catalogTable.getSchema().getPrimaryKey();
-    Assert.assertTrue(
-        "Should have the expected unique constraint", uniqueConstraintOptional.isPresent());
-    Assert.assertEquals(
-        "Should have the expected columns",
-        ImmutableList.of("key"),
-        uniqueConstraintOptional.get().getColumns());
+    assertThat(uniqueConstraintOptional.isPresent()).isTrue();
```
Comment (nastra) [inline flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/TestFlinkCatalogTable.java:136]:
```suggestion
    assertThat(uniqueConstraintOptional).isPresent();
```

File: flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/TestFlinkCatalogTable.java [hunk 2]
```diff
@@ -94,84 +89,77 @@ public void testGetTable() {
...
+    assertThat(uniqueConstraintOptional.get().getColumns())
```
Comment (nastra) [inline flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/TestFlinkCatalogTable.java:137]:
```suggestion
    assertThat(uniqueConstraintOptional.get().getColumns()).containsExactly("key");
```

File: flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/TestFlinkCatalogTable.java [hunk 3]
```diff
@@ -94,84 +89,77 @@ public void testGetTable() {
...
+                table.schema().findField("data").fieldId()));
     CatalogTable catalogTable = catalogTable("tl");
     Optional<UniqueConstraint> uniqueConstraintOptional = catalogTable.getSchema().getPrimaryKey();
-    Assert.assertTrue(
-        "Should have the expected unique constraint", uniqueConstraintOptional.isPresent());
-    Assert.assertEquals(
-        "Should have the expected columns",
-        ImmutableSet.of("data", "id"),
-        ImmutableSet.copyOf(uniqueConstraintOptional.get().getColumns()));
+    assertThat(uniqueConstraintOptional.isPresent()).isTrue();
```
Comment (nastra) [inline flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/TestFlinkCatalogTable.java:156]: same as above

File: flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/TestFlinkCatalogTable.java [hunk 4]
```diff
@@ -523,67 +499,65 @@ public void testAlterTableModifyColumnPosition() {
...
-    Assert.assertEquals(ImmutableSet.of(), schemaBefore.identifierFieldNames());
-
+    assertThat(schemaBefore.asStruct())
+        .isEqualTo(
+            new Schema(
+                    Types.NestedField.required(1, "id", Types.LongType.get()),
+                    Types.NestedField.required(2, "dt", Types.StringType.get()),
+                    Types.NestedField.optional(3, "col1", Types.StringType.get()))
+                .asStruct());
+    // Assert.assertEquals(ImmutableSet.of(), schemaBefore.identifierFieldNames());
```
Comment (nastra) [inline flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/TestFlinkCatalogTable.java:535]: please remove this

File: flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/TestFlinkCatalogTable.java [hunk 5]
```diff
@@ -523,67 +499,65 @@ public void testAlterTableModifyColumnPosition() {
...
+    assertThat(schemaBefore.identifierFieldNames()).isEmpty();
     sql("ALTER TABLE tl ADD (PRIMARY KEY (id) NOT ENFORCED)");
     Schema schemaAfterAdd = table("tl").schema();
-    Assert.assertEquals(ImmutableSet.of("id"), schemaAfterAdd.identifierFieldNames());
-
+    assertThat(schemaAfterAdd.identifierFieldNames()).isEqualTo(ImmutableSet.of("id"));
```
Comment (nastra) [inline flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/TestFlinkCatalogTable.java:539]:
```suggestion
    assertThat(schemaAfterAdd.identifierFieldNames()).containsExactly("id");
```

Comment (nastra) [reply flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/TestFlinkCatalogTable.java:539]: please also update the other places

File: flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/source/TestFlinkMetaDataTable.java [hunk 6]
```diff
@@ -164,21 +163,22 @@ public void testHistory() {
...
-          ((Instant) row.getField(0)).toEpochMilli(),
-          next.timestampMillis());
-      Assert.assertEquals("Should have expected snapshot id", next.snapshotId(), row.getField(1));
-      Assert.assertEquals("Should have expected parent id", next.parentId(), row.getField(2));
-
-      Assert.assertEquals(
-          "Should have expected is current ancestor",
-          SnapshotUtil.isAncestorOf(table, table.currentSnapshot().snapshotId(), next.snapshotId()),
-          row.getField(3));
+      assertThat(((Instant) row.getField(0)).toEpochMilli())
```
Comment (nastra) [inline flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/source/TestFlinkMetaDataTable.java:166]: I'm confused about what exactly is the actual vs expected. `testSnapshots()` has the same issue

File: flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/source/TestFlinkMetaDataTable.java [hunk 7]
```diff
@@ -189,32 +189,32 @@ public void testManifests() {
...
-          row.getField(5));
-      Assert.assertEquals(
-          "Should have expected existing_data_files_count",
-          manifestFile.existingFilesCount(),
-          row.getField(6));
-      Assert.assertEquals(
-          "Should have expected deleted_data_files_count",
-          manifestFile.deletedFilesCount(),
-          row.getField(7));
+      assertThat(manifestFile.content().id())
```
Comment (nastra) [inline flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/source/TestFlinkMetaDataTable.java:192]: shouldn't this be the `expected` part?

File: flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/source/TestFlinkMetaDataTable.java [hunk 8]
```diff
@@ -318,51 +318,50 @@ public void testUnPartitionedTable() throws IOException {
...
+    assertThat(expectedFiles).as("Should have 3 files manifest entriess").hasSize(3);
     TestHelpers.assertEquals(filesTableSchema, expectedFiles.get(0), actualFiles.get(0));
     TestHelpers.assertEquals(filesTableSchema, expectedFiles.get(1), actualFiles.get(1));
   }
 
-  @Test
+  @TestTemplate
   public void testPartitionedTable() throws Exception {
-    Assume.assumeFalse(!isPartition);
+    assumeThat(!isPartition).isFalse();
```
Comment (nastra) [inline flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/source/TestFlinkMetaDataTable.java:340]:
```suggestion
    assumeThat(isPartition).isTrue();
```

File: flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/TestFlinkCatalogTable.java (hunk 1 above)
Comment (vinitpatni) [reply flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/TestFlinkCatalogTable.java:136]: ack

File: flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/TestFlinkCatalogTable.java (hunk 3 above)
Comment (vinitpatni) [reply flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/TestFlinkCatalogTable.java:156]: ack

File: flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/TestFlinkCatalogTable.java (hunk 4 above)
Comment (vinitpatni) [reply flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/TestFlinkCatalogTable.java:535]: ack

File: flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/TestFlinkCatalogTable.java (hunk 2 above)
Comment (vinitpatni) [reply flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/TestFlinkCatalogTable.java:137]: ack

File: flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/TestFlinkCatalogTable.java (hunk 5 above)
Comment (vinitpatni) [reply flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/TestFlinkCatalogTable.java:539]: ack

File: flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/source/TestFlinkMetaDataTable.java (hunk 8 above)
Comment (vinitpatni) [reply flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/source/TestFlinkMetaDataTable.java:340]: ack

File: flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/source/TestFlinkMetaDataTable.java (hunk 6 above)
Comment (vinitpatni) [reply flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/source/TestFlinkMetaDataTable.java:166]: I fixed the order

**Comment by vinitpatni** [comment]
- Adding Junit 5 conversion and AssertJ style for TestFlinkUpsert and TestRewriteDataFilesAction

**Comment by vinitpatni** [comment]
@nastra TestStreamScanSql is the only subclass remaining for conversion to Junit 5 but it has dependency on GenericAppenderHelper class which is part of iceberg-data module. Let me know how to tackle that ?

**Comment by nastra** [comment]
@vinitpatni the issue with `GenericAppenderHelper` has been addressed in https://github.com/apache/iceberg/pull/9185, so this class should now be usable with JUnit4 + JUni5

**Comment by vinitpatni** [comment]
I have done changes in TestStreamScanSql locally but running testcases on my local machine is taking lot of time. It is running indefinitely for this class. @nastra Do you have any idea ?

**Comment by nastra** [comment]
> I have done changes in TestStreamScanSql locally but running testcases on my local machine is taking lot of time. It is running indefinitely for this class. @nastra Do you have any idea ?

@vinitpatni without seeing the changes for `TestStreamScanSql` it's difficult to say what's wrong there. Can you please push your changes for this test? The changes in the other files LGTM

File: flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/source/TestStreamScanSql.java [hunk 9]
```diff
@@ -33,30 +36,27 @@
...
+public class TestStreamScanSql extends CatalogTestBase {
   private static final String TABLE = "test_table";
   private static final FileFormat FORMAT = FileFormat.PARQUET;
 
   private TableEnvironment tEnv;
 
-  public TestStreamScanSql(String catalogName, Namespace baseNamespace) {
-    super(catalogName, baseNamespace);
-  }
+  private @TempDir Path temp;
```
Comment (nastra) [inline flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/source/TestStreamScanSql.java:59]: no need to define this, you should be able to use `temporaryDirectory` from the super class. However, I noticed that you might need to change the visibility of `Testbase.temporaryDirectory` to protected

File: flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/source/TestStreamScanSql.java [hunk 10]
```diff
@@ -127,20 +127,16 @@ private void insertRows(Table table, Row... rows) throws IOException {
 
   private void assertRows(List<Row> expectedRows, Iterator<Row> iterator) {
     for (Row expectedRow : expectedRows) {
-      Assert.assertTrue("Should have more records", iterator.hasNext());
-
+      assertThat(iterator.hasNext()).isTrue();
```
Comment (nastra) [inline flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/source/TestStreamScanSql.java:130]:
```suggestion
      assertThat(iterator).hasNext();
```

**Review by nastra — COMMENTED** [review]
almost ready to go, just 2 minor things to address

File: flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/source/TestStreamScanSql.java (hunk 9 above)
Comment (vinitpatni) [reply flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/source/TestStreamScanSql.java:59]: ack

File: flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/source/TestStreamScanSql.java (hunk 10 above)
Comment (vinitpatni) [reply flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/source/TestStreamScanSql.java]: ack

**Comment by vinitpatni** [comment]
- Addressing Review Comments

**Review by nastra — APPROVED** [review]
LGTM, thanks @vinitpatni. Could you also please remove `FlinkCatalogTestBase` as part of this PR as I don't think it's used anymore, since everything was converted.

**Comment by vinitpatni** [comment]
> LGTM, thanks @vinitpatni. Could you also please remove `FlinkCatalogTestBase` as part of this PR as I don't think it's used anymore, since everything was converted.

Ack. Removed FlinkCatalogTestBase and its references

**Review by nastra — APPROVED** [review]
thanks @vinitpatni for getting this done

**Pull request merged**